File: packages/urdu-js/src/string-utils/pad.ts
export function pad(text: string, length: number, char = ' ', direction: 'start' | 'end' = 'start'): string {
  const needed = length - [...text].length
  if (needed <= 0) return text
  const padding = char.repeat(needed)
  return direction === 'start' ? padding + text : text + padding
}

File: packages/urdu-dart/lib/src/string_utils/pad.dart
/// Pads text to the given length.
String pad(String text, int length,
    [String char = ' ', String direction = 'start']) {
  if (text.length >= length) return text;
  final padding = char * (length - text.length);
  return direction == 'end' ? '$text$padding' : '$padding$text';
}
